src/lib/dataviews/category/Category.ts

```diff
@@ -0,0 +1,57 @@
+import { Layer, Source } from '@/viz';
+import { DataViewLocal } from '../mode/DataViewLocal';
+import { DataViewRemote } from '../mode/DataViewRemote';
+import { DataViewModeAlias } from '../mode/DataViewModeBase';
+import { DataViewWrapperBase } from '../DataViewWrapperBase';
+import { CategoryOptions, CategoryImpl } from './CategoryImpl';
+
+export class Category extends DataViewWrapperBase {
+  protected buildImpl(
+    dataSource: Layer | Source,
+    column: string,
+    options: CategoryOptions,
+    mode: DataViewModeAlias
+  ) {
+    let dataView;
+
+    switch (mode) {
+      case DataViewModeAlias.NON_PRECISE: {
```

Comment:
We still need to decide on 'mode' name, we have to aggree asap on a naming, to apply it internally. After all this time, honestly I would just prefer to use something that matches reality, and internal naming, like 'Local', vs 'Server' mode, and explain via docs its differences on quality, speed and strategy for calculation

Reply:
`mode` renamed:

- `GLOBAL` to `REMOTE`
- `NON_PRECISE` to `LOCAL`
- `VIEWPORT` to `REMOTE_FILTERED` (we have to discuss about this. Not definitive)